lib/Scheduler/sched_region.cpp

```diff
@@ -77,54 +74,49 @@ void SchedRegion::CmputAbslutUprBound_() {
     const BLOCKS_TO_KEEP blocksToKeep) {
   ConstrainedScheduler *lstSchdulr = NULL;
-  ConstrainedScheduler *acoSchdulr = NULL;
+  InstSchedule *InitialSchedule = nullptr;
   InstSchedule *lstSched = NULL;
-  InstSchedule *acoSchedule = NULL;
-  llvm::opt_sched::InstCount ACOScheduleLength; //TODO: CHIPPIE: Initialization value(s)? //TODO: Chippie: What is up with the similarly-named variable in the header file? hurstcSchedLngth_; in sched_region.h, but it's passed as a parameter here. Based on the context, I'm assuming ACO's schedule length and cost should instead be added as arguments to this function?
-  InstCount ACOScheduleCost;
-  bool isACOOptimal = false; //TODO: CHIPPIE: Where should this go? Should it become an additional parameter? Why is isLstOptml a reference parameter? Or don't add it as a parameter -- just rename the existing one to "isInitialScheduleOptimal" or something? Or should they all be 3 reference parameters???
-  InstSchedule *initialSchedule = NULL;
-  bool initialScheduleOptimal = false;
-  llvm::opt_sched::InstCount initialScheduleLength; //TODO: CHIPPIE: Initialization value(s)?
-  InstCount initialScheduleCost;
+  InstSchedule *AcoSchedule = nullptr;
+  InstCount InitialScheduleLength = 0;
+  InstCount InitialScheduleCost = 0;
   FUNC_RESULT rslt = RES_SUCCESS;
   Milliseconds hurstcTime = 0;
   Milliseconds boundTime = 0;
   Milliseconds enumTime = 0;
   Milliseconds vrfyTime = 0;
-  Milliseconds acoTime = 0;
-  Milliseconds acoStart = 0;
-  Milliseconds boundStart;
+  Milliseconds AcoTime = 0;
+  Milliseconds AcoStart = 0;
 
   enumCrntSched_ = NULL;
   enumBestSched_ = NULL;
   bestSched = bestSched_ = NULL;
 
-  //TODO: CHIPPIE: Perhaps rename these to match the flag names?
-  bool heuristicSchedulerEnabled = SchedulerOptions::getInstance().GetBool("HEUR_ENABLED");
-  bool acoSchedulerEnabled = SchedulerOptions::getInstance().GetBool("ACO_ENABLED");
-  bool bbSchedulerEnabled = SchedulerOptions::getInstance().GetBool("BB_ENABLED");
+  bool AcoBeforeEnum = false;
+  bool AcoAfterEnum = false;
 
-  if (false == heuristicSchedulerEnabled && false == acoSchedulerEnabled) {
-    //Abort if ACO and heuristic algorithms are disabled.
-    cout << "TODO: Descriptive error message here saying that there must be at least one of the ACO or Heuristic scheduler enabled.\n";
-    return RES_ERROR;
+  Config &schedIni = SchedulerOptions::getInstance();
+  bool HeuristicSchedulerEnabled = schedIni.GetBool("HEUR_ENABLED");
+  bool AcoSchedulerEnabled = schedIni.GetBool("ACO_ENABLED");
+  bool BbSchedulerEnabled = schedIni.GetBool("ENUM_ENABLED");
+  if (AcoSchedulerEnabled) {
+    AcoBeforeEnum = schedIni.GetBool("ACO_BEFORE_ENUM");
+    AcoAfterEnum = schedIni.GetBool("ACO_AFTER_ENUM");
   }
 
-  /*
-    * Algorithm run order:
-    * 1) Heuristic
-    * 2) ACO
-    * 3) Branch & Bound
-    * 
-    * Each of these 3 algorithms can be individually disabled, but one of the heuristic or ACO must be enabled. (TODO: CHIPPIE)
-    */
```

Comment:
Would it be a good idea to update this block comment and keep it at the top of the function as a high-level overview? Might be helpful (especially for newcomers) to have an at-a-glance overview of the order the function does major functionality. It is a fairly large function...

Reply:
I agree this is a large function but I am not certain if this belongs in the cpp file. What do you think about the header file?

Reply:
I think that'd work too, but I generally find it more appropriate to put implementation related comments in the cpp file and mainly interface comments/usage in the header file.

Reply:
That's a good point. I have added it back to the cpp file.